extra_data/reader.py

```diff
@@ -593,17 +593,33 @@ def union(self, *others):
         files = set(self.files)
         train_ids = set(self.train_ids)
 
+        run_num = self.run_metadata()[
+            "runNumber"] if self.is_single_run else None
+
+        proposal_num = self.run_metadata()[
+            "proposalNumber"] if self.is_single_run else None
+
         for other in others:
             files.update(other.files)
             train_ids.update(other.train_ids)
+            # Set self.is_single_run to False
+            # in case of Unioning multiple run DataCollections.
+
+            if not (
+                other.is_single_run and
+                other.run_metadata()["runNumber"] == run_num and
+                other.run_metadata()["proposalNumber"] == proposal_num
+            ):
+                self.is_single_run = False
```

Comment:
This shouldn't modify `self` - it should only affect the new DataCollection we return.

Reply:
yes, true. Fixed it.